src/native-common/View.tsx

```diff
@@ -343,17 +343,20 @@ export class View extends ViewBase<Types.ViewProps, Types.Stateless> {
             accessibilityComponentType: AccessibilityUtil.accessibilityComponentTypeToString(props.accessibilityTraits),
             accessibilityLiveRegion: AccessibilityUtil.accessibilityLiveRegionToString(props.accessibilityLiveRegion)
         };
-        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation && (this.props.onPress ||
-                (this.props.tabIndex !== undefined && this.props.tabIndex >= 0))) {
+
+        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation) {
             const macAccessibilityProps: MacComponentAccessibilityProps = accessibilityProps as any;
-            if (this.props.tabIndex !== -1) {
+
+            if (this.props.tabIndex !== undefined && this.props.tabIndex >= 0) {
```

Comment:
This is still a change in behaviour.  If tabIndex is undefined and onPress is defined, the below code should be run.
You're going to end up with the same conditional that was above